elasticdl/python/common/k8s_volume.py

```diff
@@ -1,7 +1,71 @@
-_ALLOWED_VOLUME_KEYS = ["claim_name", "mount_path"]
+from kubernetes import client
+
+_ALLOWED_VOLUME_KEYS = ["claim_name", "host_path", "type", "mount_path"]
+
+
+def parse_volume_and_mount(volume_conf, pod_name):
+    """Get k8s volumes list and volume mounts list from
+    the volume config string.
+
+    Args:
+        volume_conf (string): the volumes config string,
+        e.g. "host_path=c0,mount_path=/path0;claim_name=c1,mount_path=/path1".
+        pod_name (string): the pod name
+
+    Return:
+        volumes (List): a Python list contains k8s volumes.
+        volume_mounts (List): a Python list contains k8s volume mounts.
+    """
+    volumes = []
+    volume_mounts = []
+    volume_dicts = parse(volume_conf)
+    for i, volume_dict in enumerate(volume_dicts):
+        volume_name = pod_name + "-volume-%d" % i
+        if "claim_name" in volume_dict:
+            pvc_volume_source = client.V1PersistentVolumeClaimVolumeSource(
+                claim_name=volume_dict["claim_name"], read_only=False
+            )
+            volume = client.V1Volume(
+                name=volume_name, persistent_volume_claim=pvc_volume_source
+            )
+        elif "host_path" in volume_dict:
+            volume = client.V1Volume(
+                name=volume_name,
+                host_path=client.V1HostPathVolumeSource(
+                    path=volume_dict["host_path"],
+                    type=volume_dict.get("type", None),
+                ),
+            )
+        volumes.append(volume)
+        volume_mounts.append(
+            client.V1VolumeMount(
+                name=volume_name, mount_path=volume_dict["mount_path"]
+            )
+        )
+    return volumes, volume_mounts
 
 
 def parse(volume_str):
+    """Parse combined k8s volume strings separated by
+    semicolons to Python dictionaries.
```

Comment:
to a list of python dictionaries

Reply:
We may declare that the result is an array instead of dict

Reply:
The result has been declared as a list in `Return`  of the docstring.